src/Bicep.Decompiler/BicepparamDecompiler.cs

```diff
@@ -0,0 +1,200 @@
+        {
+            throw new InvalidOperationException($"Failed to read {entryJsonUri}");
+        }
+
+        var program = DecompileParamFile(jsonInput, entryBicepparamUri, bicepFilePath);
+
+        var bicepparamFile = SourceFileFactory.CreateBicepParamFile(entryBicepparamUri, program.ToText());
+
+        workspace.UpsertSourceFile(bicepparamFile);
+
+        return new DecompileResult(entryBicepparamUri, PrintFiles(workspace));
+    }
+
+    public ProgramSyntax DecompileParamFile(string jsonInput, Uri entryBicepparamUri, string? bicepFilePath)
+    {
+        var statements = new List<SyntaxBase>();
+
+        var jsonReader = new JsonTextReader(new StringReader(jsonInput));
+
+        JObject jsonObject = JObject.Load(jsonReader, new JsonLoadSettings{
+            CommentHandling = CommentHandling.Ignore,
+            LineInfoHandling = LineInfoHandling.Load
+        });
+
+        statements.Add(new UsingDeclarationSyntax(
+            SyntaxFactory.CreateIdentifierToken("using"), 
+            bicepFilePath is { } ?
+            SyntaxFactory.CreateStringLiteral(bicepFilePath): 
+            SyntaxFactory.CreateStringLiteralWithComment("", "TODO: Provide a path to a bicep template")));        
+
+            statements.Add(SyntaxFactory.DoubleNewlineToken);
+
+
+        var parameters = (TemplateHelpers.GetProperty(jsonObject, "parameters")?.Value as JObject ?? new JObject()).Properties();        
+
+        foreach(var parameter in parameters)
+        {
+            var metadata = parameter.Value?["metadata"];
+
+            if(metadata is {})
+            {
+                statements.Add(ParseParameterWithComment(metadata));
+                statements.Add(SyntaxFactory.NewlineToken);
+            }
+
+            statements.Add(ParseParam(parameter));
+            statements.Add(SyntaxFactory.DoubleNewlineToken);
+        }
+
+        var program = new ProgramSyntax(statements, SyntaxFactory.CreateToken(Core.Parsing.TokenType.EndOfFile));
+
+        return program;
+    }
+
+    private SyntaxBase ParseParam(JProperty param)
+    { 
+        if(param.Value?["reference"] is {})
+        {
+            return new ParameterAssignmentSyntax(
+            SyntaxFactory.CreateIdentifierToken("param"),
+            SyntaxFactory.CreateIdentifier(param.Name),
+            SyntaxFactory.AssignmentToken,
+            SyntaxFactory.CreateInvalidSyntaxWithComment("KeyVault references are not supported in Bicep Parameters files"));
+        }
+
+        var value = param.Value?["value"];
+
+        if(value is null)
+        {
+            throw new Exception($"No value found parameter {param.Name}");
+        }
+
+        return new ParameterAssignmentSyntax(
+            SyntaxFactory.CreateIdentifierToken("param"),
+            SyntaxFactory.CreateIdentifier(param.Name),
+            SyntaxFactory.AssignmentToken,
+            ParseJToken(value)
+        );
+    }
+
+    private SyntaxBase ParseJToken(JToken value)
```

Comment:
For these ParseJ_ methods like ParseJToken, is there not a way to reuse some of the existing methods we have?

Reply:
We do have some similar methods in Bicep decompiler that we can technically reuse but the concern is that over time the decompilation logic in bicep and bicepparam would diverge such that using common method may not make sense